sdk/src/wallet/update.rs

```diff
@@ -0,0 +1,211 @@
+            }
+        }
+
+        #[cfg(feature = "storage")]
+        {
+            log::debug!("[SYNC] storing wallet {} with new synced data", wallet_data.alias);
+            self.save(Some(&wallet_data)).await?;
+        }
+        Ok(())
+    }
+
+    /// Update wallet with newly synced transactions.
+    pub(crate) async fn update_with_transactions(
+        &self,
+        updated_transactions: Vec<TransactionWithMetadata>,
+        spent_output_ids: Vec<OutputId>,
+        output_ids_to_unlock: Vec<OutputId>,
+    ) -> crate::wallet::Result<()> {
+        log::debug!("[SYNC] Update wallet with new synced transactions");
+
+        let mut wallet_data = self.data_mut().await;
+
+        for transaction in updated_transactions {
+            match transaction.inclusion_state {
+                InclusionState::Confirmed | InclusionState::Conflicting | InclusionState::UnknownPruned => {
+                    let transaction_id = transaction.payload.transaction().id();
+                    wallet_data.pending_transactions.remove(&transaction_id);
+                    log::debug!(
+                        "[SYNC] inclusion_state of {transaction_id} changed to {:?}",
+                        transaction.inclusion_state
+                    );
+                    #[cfg(feature = "events")]
+                    {
+                        self.emit(WalletEvent::TransactionInclusion(TransactionInclusionEvent {
+                            transaction_id,
+                            inclusion_state: transaction.inclusion_state,
+                        }))
+                        .await;
+                    }
+                }
+                _ => {}
+            }
+            wallet_data
+                .transactions
+                .insert(transaction.payload.transaction().id(), transaction.clone());
+        }
+
+        for output_to_unlock in &spent_output_ids {
+            if let Some(output) = wallet_data.outputs.get_mut(output_to_unlock) {
+                output.is_spent = true;
+            }
+            wallet_data.locked_outputs.remove(output_to_unlock);
+            wallet_data.unspent_outputs.remove(output_to_unlock);
+            log::debug!("[SYNC] Unlocked spent output {}", output_to_unlock);
+        }
+
+        for output_to_unlock in &output_ids_to_unlock {
+            wallet_data.locked_outputs.remove(output_to_unlock);
+            log::debug!(
+                "[SYNC] Unlocked unspent output {} because of a conflicting transaction",
+                output_to_unlock
+            );
+        }
+
+        #[cfg(feature = "storage")]
+        {
+            log::debug!(
+                "[SYNC] storing wallet {} with new synced transactions",
+                wallet_data.alias
+            );
+            self.save(Some(&wallet_data)).await?;
+        }
+        Ok(())
+    }
+
+    /// Update the wallet address with a possible new Bech32 HRP and clear the inaccessible_incoming_transactions.
```

Comment:
```suggestion
    /// Update the wallet address with a possible new Bech32 HRP and clear the inaccessible incoming transactions.
```

Reply:
or maybe not ... since it's just a doc comment. I'll commit the suggestion.